ITE/10/README.adoc

```diff
@@ -0,0 +1,287 @@
+the name of the step and the key ID used to sign it. This part of the workflow
+is updated to support matching on predicate types as well. In addition,
+artifact rules are updated to support
+link:https://github.com/in-toto/attestation/blob/main/spec/v1/resource_descriptor.md[ResourceDescriptor]
+defined in the in-toto Attestation Framework. The `name` field in a
+ResourceDescriptor object is used for pattern matching and the `digests` field
+is used to verify equality of two artifacts.
+
+```go
+func Verify(layoutEnvelope, layoutKeys, attestations, now) {
+    verifyLayoutSignatures(layoutEnvelope, layoutKeys)
+
+    layout := layoutEnvelope.Payload
+    verifyLayoutExpiry(layout, now)
+
+    if layout.Steps == nil && layout.Inspections == nil {
+        panic
+    }
+
+
+    for _, step := range layout.Steps {
+        stepAttestations := attestationsForStep(step.Name, attestations)
+        for _, predicate := range step.ExpectedPredicates {
+            predicateAttestations := attestationsForStepPredicate(predicate.Type, stepAttestations)
+            if predicate.Type == LAYOUT {
+                Verify(predicateAttestations[0], predicate.Functionaries, predicateAttestations[1:], now)
+            } else {
+                verifiedClaims := verifyAttestationSignatures(predicate.Functionaries, predicateAttestations)
+		if len(verifiedClaims) < threshold {
+		    panic
+		}
+                for _, attestation := range verifiedClaims {
+                    applyMaterialRules(step.ExpectedMaterials, attestation.Materials, attestations)
+                    applyProductRules(predicate.ExpectedProducts, attestation.Products, attestations)
+                }
+            }
+        }
+    }
+
+    for _, inspection := range layout.Inspections {
+        inspectionAttestation := executeInspection(inspection.Command)
+        applyMaterialRules(inspection.ExpectedMaterials, inspectionAttestation.Materials, attestations)
+        applyProductRules(inspection.ExpectedProducts, inspectionAttestation.Products, attestations)
+    }
+}
+```
+
+=== Custom Verifiers
+
+As noted above, in-toto v1.0's verification workflow generates an in-toto
+summary link if verification succeeds. This is a powerful semantic that allows
+features like sublayouts that allow for delegating the supply chain definition
+for a step to authorized functionaries, as well as summarized verification for
+environments with limited resources. With the introduction of in-toto
+Attestation Framework, the verification workflow can be parametrized to support
+generating specific predicates as needed. For example, the default verifier can
+generate an in-toto link, another can generate SLSA Provenance, while for the
+summary use cases, a third verifier type can generate Verification Summary
+Attestations (VSAs).
+
+Each of these verifiers must be well defined with respect to the predicates they
+generate. Also, each verifier must perform the basic in-toto verification
```

Comment:
Can we be a bit more precise here? Not sure what "well defined with respect to the predicates" means. And is "basic in-toto verification" actually referring to the in-toto 1.0 verification flow?

Reply:
Restated this a bit.